app/models/miq_queue.rb

```diff
@@ -623,6 +623,16 @@ def self.display_name(number = 1)
 
   private
 
+  # NOTE: this will intentionally lookup missing zones every time
+  def validate_zone_name
+    if zone && !self.class.valid_zone_names[zone]
```

Comment:
ah, how do we want to handle caching false?

I know it's unlikely, but won't invalid zones not be cached?  I believe invalid zones will never be cached.

I don't know if that is intentional or not.

Reply:
I'm fine with this but wasn't sure if this was the intention.

Reply:
that was intentional. Jason wanted to cover the case that a newly zone was created.
Since that is a rare edge case, I felt it was ok to incur a cost.

The main case I want to optimize is using the same zone (typically the default zone but basically the ems zone) over and over again.

Reply:
Thanks.  Sorry, I missed that explanation.  Yes, I agree.